lib/quake_parser.ex

```diff
@@ -3,42 +3,120 @@ defmodule QuakeParser do
   `QuakeParser` is a parser for Quake game logs.
   """
 
-  alias QuakeParser.{Game, FindPlayers, KillsOfPlayer}
+  defstruct total_kills: 0, players: [], kills: %{}
 
+  @doc """
+  Parse the Quake log file, and return a QuakeParser object for each game.
+  """
+  @spec start(String.t()) :: list()
   def start(path) do
-    validate_file(path)
-    |> find_games
-    |> Enum.map(&parse_game/1)
+    with {:ok, content} <- File.read(path) do
+      content
+      |> find_games
+      |> Enum.map(&parse_game/1)
+    end
+  end
+
+  @doc """
+  Parse the Quake log file, and return a map with the number of game in a row, and each value
+  is another map with how much deaths per meaning.
+  """
+  @spec death_report(String.t()) :: Map.t()
+  def death_report(path) do
+    with {:ok, content} <- File.read(path) do
+      content
+      |> find_games
+      |> Enum.map(&game_kills/1)
+      |> Enum.map(&build_death_report/1)
+      |> Enum.with_index()
+      |> Enum.map(fn {obj, index} -> {index, obj} end)
+      |> Enum.into(%{})
+    end
   end
 
   defp parse_game(game) do
     total_kills = game_kills(game) |> Enum.count()
-    players = FindPlayers.call(game)
-    kills = kills_of_each_player(players, game)
+    players = find_players(game)
+    kills = count_death_by_players(players, game)
 
-    %Game{total_kills: total_kills, players: players, kills: kills}
+    %__MODULE__{total_kills: total_kills, players: players, kills: kills}
   end
 
   defp find_games(log_content) do
     [_h | games] = String.split(log_content, "InitGame:")
 
-    games |> Enum.map(fn str -> String.split(str, "\n") end)
+    Enum.map(games, &String.split(&1, "\n"))
   end
 
-  defp kills_of_each_player(players, game) do
+  defp find_players(game) do
+    key = "ClientUserinfoChanged:"
+    start = " n\\"
+    finish = "\\t"
+
+    game
+    |> Enum.filter(fn str -> String.contains?(str, key) end)
+    |> Enum.map(fn str -> String.split(str, start) end)
+    |> Enum.map(fn [_, str] -> str end)
+    |> Enum.map(fn str -> String.split(str, finish) end)
+    |> Enum.map(fn [str | _] -> str end)
+    |> Enum.map(fn str -> String.replace(str, "!", "") end)
+    |> Enum.uniq()
+  end
+
+  defp count_death_by_players(players, game) do
     players
-    |> Enum.map(fn player -> KillsOfPlayer.call(player, game_kills(game)) end)
+    |> Enum.map(&kills_of_player(&1, game_kills(game)))
     |> Enum.into(%{})
   end
 
   defp game_kills(game) do
```

Comment:
```suggestion
  defp game_kills(game), do: Enum.filter(game, &String.contains?(&1, "Kill:"))
```

Da pra deixar inline. E tb, `"Kill:")` talvez ficasse mais claro sobre o q é se declarasse como uma module variable no topo `@minha_var "Kill:"`